core/src/main/java/org/apache/accumulo/core/summary/SummaryReader.java

```diff
@@ -177,7 +179,11 @@ private static SummaryReader load(CachableBlockFile.Reader bcReader,
   public static SummaryReader load(Configuration conf, RFileSource source, String cacheId,
       Predicate<SummarizerConfiguration> summarySelector, SummarizerFactory factory,
       CryptoService cryptoService) throws IOException {
-    CachableBuilder cb = new CachableBuilder().input(source.getInputStream(), cacheId)
+    InputStream input = source.getInputStream();
+    if (!(input instanceof FSDataInputStream)) {
+      throw new IllegalArgumentException("Input stream is not a FSDataInputStream");
+    }
```

Comment:
If this code were omitted, then the cast would fail with an exception if its not an instance of.  Why not just let that fail?

Reply:
Removed instanceof checks in 0957d10